### PR TITLE
WELD-2445 Allow to recognize beans 1.1 xsd.

### PR DESCRIPTION
**Pull request by manovotn**
I don't like it much but it's the best idea we came up with ATM.

Also, I am still fishing for actual test for this.

### CODE REVIEW

**Comment by manovotn** [comment]
BTW I tried the semi-test of using `beans.xml` with version 1.1 and using `<trim/>` at the same time. But Weld will simply log error and process the file anyway (handling it as trimmed archive). I am kinda out of ideas for tests.

File: impl/src/main/java/org/jboss/weld/xml/BeansXmlParser.java [hunk 1]
```diff
@@ -178,11 +196,12 @@ private boolean alreadyAdded(Metadata<String> metadata, List<Metadata<String>> l
...
     }
 
-    private static InputSource[] loadXsds() {
+    private static InputSource[] loadXsds(String beansXmlAsString) {
 
         List<InputSource> xsds = new ArrayList<InputSource>();
+        String beans11 = "beans_1_1.xsd";
 
-        for (XmlSchema schema : XmlSchema.values()) {
+        for (XmlSchema schema : XmlSchema.getSchemas(beansXmlAsString != null && beansXmlAsString.contains(beans11))) {
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/xml/BeansXmlParser.java]: The `beans11` variable is not needed - we could use `XmlSchema.CDI11.getFileName()` instead.

File: impl/src/main/java/org/jboss/weld/xml/XmlSchema.java [hunk 2]
```diff
@@ -16,11 +16,14 @@
...
 
+import java.util.ArrayList;
+import java.util.List;
+
 import javax.enterprise.inject.spi.BeanManager;
 
 public enum XmlSchema {
 
-    CDI10("beans_1_0.xsd", BeanManager.class.getClassLoader()), CDI20("beans_2_0.xsd", BeanManager.class.getClassLoader()), WELD11("weld_1_1.xsd", BeansXmlParser.class.getClassLoader());
+    CDI10("beans_1_0.xsd", BeanManager.class.getClassLoader()), CDI11("beans_1_1.xsd", BeanManager.class.getClassLoader()), CDI20("beans_2_0.xsd", BeanManager.class.getClassLoader()), WELD11("weld_1_1.xsd", BeansXmlParser.class.getClassLoader());
```
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/xml/XmlSchema.java]: Hm, all the values use the same CL, maybe we could remove the arg and use `BeanManager.class.getClassLoader()` by default?

Comment (manovotn) [reply impl/src/main/java/org/jboss/weld/xml/XmlSchema.java]: `WELD11` uses different CL, so I kept it as is.

**Pull request merged**